dockstore-webservice/src/main/java/io/dockstore/webservice/languages/WDLHandler.java

```diff
@@ -62,18 +63,56 @@
     public static final String ERROR_PARSING_WORKFLOW_YOU_MAY_HAVE_A_RECURSIVE_IMPORT = "Error parsing workflow. You may have a recursive import.";
     private static final Pattern IMPORT_PATTERN = Pattern.compile("^import\\s+\"(\\S+)\"");
 
+    public static void checkForRecursiveLocalImports(String content, Set<SourceFile> sourceFiles, Set<String> localFilePaths, String parent)
+            throws Exception {
+        // Use matcher to get imports
+        String[] lines = StringUtils.split(content, '\n');
+        for (String line : lines) {
+            Matcher m = IMPORT_PATTERN.matcher(line);
+
+            while (m.find()) {
+                String match = m.group(1);
+                if (!match.startsWith("http://") && !match.startsWith("https://")) { // Don't resolve URLs
+                    String localImport = match.replaceFirst("file://", "");
+                    if (localFilePaths.contains(localImport)) {
```

Comment:
I'm worried this could run into the same problem that we ran into parsing gatk-sv, where they have the different files with the same name in different directories. Do you not need to have the absolute path in the Set?

They have something like `import 'Struts.wdl'`, in many different files in different directories, but each directory has its own version of struts.wdl.

Reply:
I agree, I think this is a bit redundant as well. 

Can this check be moved to `processImports` so that we don't essentially look at imports twice. 

The second change is that this should add and test for a validation message as in https://github.com/dockstore/dockstore/blob/a473a6e893248f32b0d0abce603e0bd4f6080dda/dockstore-webservice/src/main/java/io/dockstore/webservice/languages/WDLHandler.java#L157

Reply:
@coverbeck It's already been tested with Structs.wdl already. Essentially, if you imagine the files are imported in a tree-like structure. The path between the root and the leaves should contain no duplicates, but the many paths may have the same file.